client/src/components/PersonForm/CompanySelect/index.js

```diff
@@ -0,0 +1,106 @@
+import './index.scss';
+import VueSelect from 'vue-select';
+import { debounce } from 'debounce';
+import { DEBOUNCE_WAIT } from '@/globals/constants';
+import formatOptions from '@/utils/formatOptions';
+import Button from '@/components/Button';
+import ErrorMessage from '@/components/ErrorMessage';
+
+// @vue/component
+export default {
+    name: 'CompanySelect',
+    props: {
+        defaultCompany: { type: Object, default: null },
+    },
+    data() {
+        const hasDefaultValue = !!this.defaultCompany;
+        const defaultItem = {
+            value: this.defaultCompany?.id || null,
+            label: this.defaultCompany?.legal_name || null,
+        };
+
+        return {
+            selectedValue: hasDefaultValue ? defaultItem : null,
+            companies: hasDefaultValue ? [defaultItem] : [],
+            searchError: null,
+        };
+    },
+    methods: {
+        // - Ici on n'utilise pas une fonction fléchée, afin de pouvoir accéder à 'this'
+        // eslint-disable-next-line func-names
+        search: debounce(async function (loading, search) {
+            try {
+                this.searchError = null;
+                const params = { search, limit: 10 };
+                const { data } = await this.$http.get('companies', { params });
+                this.companies = formatOptions(data.data, (item) => item.legal_name);
+            } catch (error) {
+                this.searchError = error;
+            } finally {
+                loading(false);
+            }
+        }, DEBOUNCE_WAIT),
```

Comment:
Malheureusement ce n'est pas safe de faire ça ainsi, si au moment ou ton component est unmounted tu as un debounce en cours, il va s'executer après que le component soit détruit. C'est pour ça que `debounce` intègre une [méthode `clear`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/debounce/index.d.ts#L11).

En gros il faudrait que tu déclare ta méthode `search` normalement et faire un truc du genre:

```js
created() {
    this.debouncedSearch = debounce(this.search.bind(this), DEBOUNCE_WAIT);
},
beforeUnmount() {
    this.debouncedSearch.clear();
},
```

Ca c'est partant du prince que `this.search` est déjà accessible dans `created` (je crois que oui mais dans le doute ...). 
Sinon il faudra directement déclarer le callback dans le created, ce serait moins propre mais bon.

Reply:
OK c'est tout bon, on peut utiliser ce qu'on veut dans le `created`, ta méthode fonctionne parfaitement :+1: !!